File: ListViews/ListDemo/app/src/main/java/edu/cs4730/ListDemo/Simple1_ListFragment.java
package edu.cs4730.ListDemo;

/*
 * much of this code was adapted from
 * http://www.vogella.de/articles/AndroidListView/article.html
 *
 */


import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.fragment.app.ListFragment;

import edu.cs4730.ListDemo.databinding.ListfragmentLayoutBinding;

public class Simple1_ListFragment extends ListFragment {

    String TAG = "Simple_frag";
    String[] values = new String[]{"Android", "iPhone", "WindowsMobile",
            "Blackberry", "WebOS", "Ubuntu", "Windows7", "Max OS X",
            "Linux", "OS/2", "VMS", "DOS", "Other", "Chrome"};
    ListfragmentLayoutBinding binding;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        binding = ListfragmentLayoutBinding.inflate(inflater, container, false);
        ArrayAdapter<String> adapter = new ArrayAdapter<String>(requireContext(), android.R.layout.simple_list_item_1, values);
        setListAdapter(adapter);
        return binding.getRoot();

    }

    @Override
    public void onListItemClick(@NonNull ListView l, @NonNull View v, int position, long id) {
        String item = values[position];   //getListAdapter().getItem(position);
        Toast.makeText(requireContext(), item + " selected", Toast.LENGTH_LONG).show();
    }

}

File: ListViews/ListDemo_kt/app/src/main/java/edu/cs4730/listdemo_kt/Simple1_ListFragment.kt
package edu.cs4730.listdemo_kt

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ListView
import android.widget.Toast
import androidx.fragment.app.ListFragment
import edu.cs4730.listdemo_kt.databinding.ListfragmentLayoutBinding

class Simple1_ListFragment : ListFragment() {
    var TAG = "Simple_frag"
    var values = arrayOf(
        "Android",        "iPhone",        "WindowsMobile",        "Blackberry",
        "WebOS",        "Ubuntu",        "Windows7",        "Max OS X",
        "Linux",        "OS/2",        "VMS",        "DOS",
        "Other",        "Chrome"
    )
    lateinit var binding: ListfragmentLayoutBinding

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        binding = ListfragmentLayoutBinding.inflate(inflater, container, false)
        val adapter = ArrayAdapter(inflater.context, android.R.layout.simple_list_item_1, values)
        listAdapter = adapter
        return binding.root
    }

    override fun onListItemClick(l: ListView, v: View, position: Int, id: Long) {
        Toast.makeText(requireContext(), "${values[position]} selected", Toast.LENGTH_LONG).show()
    }
}

